core/src/main/java/org/bitcoinj/core/Peer.java

```diff
@@ -899,14 +896,7 @@ public void startMasternodeListDownload() {
                     blockChain.getBlockStore().get(blockChain.getBestChainHeight() - SigningManager.SIGN_HEIGHT_OFFSET);
 
             if (context.masternodeListManager.getListAtChainTip().getHeight() < masternodeListBlock.getHeight()) {
-                if (!LLMQUtils.isQuorumRotationEnabled(headerChain, params, params.getLlmqForInstantSend())) {
-                    GetSimplifiedMasternodeListDiff msg = new GetSimplifiedMasternodeListDiff(context.masternodeListManager.getListAtChainTip().getBlockHash(), masternodeListBlock.getHeader().getHash());
-                    sendMessage(msg);
-                } else {
-                    GetQuorumRotationInfo msg = context.masternodeListManager.getQuorumRotationInfoRequest(headerChain.getChainHead());
-                    log.info("message = {}, {}", msg, this);
-                    sendMessage(msg);
-                }
+                context.masternodeListManager.requestQuorumStateUpdate(this, headerChain.getChainHead());
```

Comment:
This is used in DashJ many, many times (`context.masternodeListManager`) and I would like to get away from it.

The context object is effectively a singleton object.  There can only be one context object associated with a thread and only one `NetworkParameters` associated with context.  This limits how many chains an app can be working with (mainnet only, testnet only, etc)

If we wanted an app to support mainnet and testnet at the same time.

Reply:
As far as the change made here, this code before the PR is assuming that the DIP4 method of obtaining quorums would be replaced by DIP24.  That is why there is an `if` block.  However, we will need to support DIP4 and DIP24 at the same time.

Additionally, we can say that `Peer` shouldn't care which DIPx is being used to sync masternode lists.  The Masternode List Manager should handle that.